rllib/algorithms/sac/sac.py

```diff
@@ -371,3 +435,149 @@ def get_default_policy_class(
             return SACTorchPolicy
         else:
             return SACTFPolicy
+
+    @override(DQN)
+    def training_step(self) -> ResultDict:
+        # If `RolloutWorker` is used, then use as the old stack `training step`
+        # of `DQN`.
+        if not self.config.uses_new_env_runners:
+            return super().training_step()
+
+        # Alternate between storing and sampling and training.
+        store_weight, sample_and_train_weight = calculate_rr_weights(self.config)
+        train_results = {}
+
+        # Run multiple sampling iterations.
+        for _ in range(store_weight):
+            # Otherwise, we use the new stack.
+            with self._timers[SAMPLE_TIMER]:
+                if self.workers.num_remote_workers() <= 0:
```

Comment:
Actually, let's fix this for good:
Let's use the existing (old stack) `synchronous_parallel_sample()` method from rllib/execution/rollout_ops.py

We will have to slightly enhance this method to support the new stack (with EnvRunners returning episodes instead of batches). But the advantage will then be that we get a fault tolerant parallel sampling step!

We can also change PPO's training_step method then to use this utility again for the new stack.

Reply:
Alright. Shall we do this in a new PR?

Reply:
Fine with me, but let's add a TODO here, then.

Reply:
Fine with me, but let's add a TODO here, then.